src/backends/torch/torchinputconns.h

```diff
@@ -79,6 +79,20 @@ namespace dd
     {
     }
 
+    void check_tests_sizes(size_t tests_size, size_t names_size)
+    {
+      if (tests_size != names_size)
+        {
+          std::string msg
+              = "something wrong happened with datasets definitions: "
```

Comment:
When is this supposed to happen? Is it an internal error or bad parameters from the user?

Reply:
It is not supposed to happen, this smells fear

Reply:
If this is not supposed to happen it should be a `InputConnectorInternalException`

Reply:
ok, done